### PR TITLE
Ditch compile_local.py and minor improvement to run locally

### PR DESCRIPTION
**Pull request by FlorianSW**
I just started with working with the source of this mission and tried to
build it locally. However, the builddir and packed dir where not
created. One of the improvements here is now, that these directories are
created automatically if they are not present.

Also: When looking at the compile.py and compile_local.py, they loocked
pretty much identical. I merged the different things and made the
compile.py script being able to run locally and therefore
compile_local.py could be removed (which I boldly did in this PR as
well).

### CODE REVIEW

**Comment by FlorianSW** [comment]
Hope it's okay to jump newly into this repo with probably such a "big" PR :) I'm looking forward to your opinion! :-)

**Comment by FlorianSW** [comment]
Did anyone has a chance to take a look? :) Ai would also try and look if there are more improvements possible in this build script to make it easier understandable and such. For that, it would be nice to do that only in one file :-)

**Comment by NeoArmageddon** [comment]
I added a comment to line 25 in compile_local.py. Can't you see it?

FYI my long term plan was to switch to a new mission compiler, not based on python.

**Comment by AKA-TiTAN** [comment]
i currently rebuild the mission into separate server.pbo and mission.pbo. it makes the mission files very small, as alot that runs serverside is common in all missions. loadtimes etc are much better. maybe this could be another way to build it?

**Comment by NeoArmageddon** [comment]
> i currently rebuild the mission into separate server.pbo and mission.pbo. it makes the mission files very small, as alot that runs serverside is common in all missions. loadtimes etc are much better. maybe this could be another way to build it?

That was planned. The current implementation was just the fastes way to get the mission into SteamWS.

**Comment by FlorianSW** [comment]
> I added a comment to line 25 in compile_local.py. Can't you see it?

Nope, can't see it. Maybe it's pending until you submit the review? (Instead of a single comment, it's probably part of a review)

> FYI my long term plan was to switch to a new mission compiler, not based on python.

Ah, ok. That's fine, I think until then, there might still some gains improving the current way, maybe :)

File: compile_local.py [hunk 1]
```diff
@@ -1,150 +0,0 @@
...
-    print("Parsing config "+scfg)    
-    with open(scfg) as json_adata:  
-        adata = json.load(json_adata)  
-        mods = mods + adata['Mods'];
-        islands = islands + adata['Islands'];
-        missions = missions + adata['Missions'];
-        addons = addons + adata['Addons']
-#Add devbuild number to version
-#if os.environ['CI_COMMIT_REF_NAME'] == "develop": 
-data['replace']['VERSION'] += ' dev ' + now.strftime("%y%m%d %H%M")
```
[the remark this answers is not in the document]
Comment (NeoArmageddon) [reply compile_local.py:25]: Jep, that was it. No late night commenting anymore for me...

[the remark this answers is not in the document]
Comment (FlorianSW) [reply compile_local.py:25]: > Jep, that was it. No late night commenting anymore for me...

Np, you're not the first and for sure not the last one falling into this issue :D

> compile_local.py is the newer file (compile.py was used with Jenkins CI). and contains the canonical way of naming dev versions. This is missing from your merged compile.py

I'm not sure, if I understand you correctly, as L24 in compile.py should more or less create the same result (there is a whitespace missing, though). So, what exactly do you mean? :-)

[the remark this answers is not in the document]
Comment (NeoArmageddon) [reply compile_local.py:25]: I am not using a CI system anymore for compile dev builds or when testing locally, thus I replaced environment variables with a hardcoded string "dev" and the buildnumber changed to a date in form "yyyymmdd hhmm". The "build number" environment variable is not used for dev builds anymore.

[the remark this answers is not in the document]
Comment (NeoArmageddon) [reply compile_local.py:25]: > I am not using a CI system anymore for compile dev builds or when testing locally, thus I replaced environment variables with a hardcoded string "dev" and the buildnumber changed to a date in form "yyyymmdd hhmm". The "build number" environment variable is not used for dev builds anymore. 
So in short: build number environvar can be removed and actual number needs hour and minutes added to quickly discern between different builds on the same day (and to sort them).

[the remark this answers is not in the document]
Comment (FlorianSW) [reply compile_local.py:25]: We also do not need the 'COMMIT' replace anymore, isn't it?

[the remark this answers is not in the document]
Comment (NeoArmageddon) [reply compile_local.py:25]: Yes.

**Pull request merged**